openstack/sharedfilesystems/v2/shares/results.go

```diff
@@ -110,3 +110,39 @@ type DeleteResult struct {
 type GetResult struct {
 	commonResult
 }
+
+// GrantAccessRes contains all information associated with an OpenStack share Grant Access
+type GrantAccessRes struct {
+	// The UUID of the share to which you are granted or denied access.
+	ShareID string `json:"share_id"`
+	// Timestamp when the share was created
+	CreatedAt time.Time `json:"created_at, omitempty"`
+	// Timestamp when the share was updated
+	UpdatedAt time.Time `json:"updated_at, omitempty"`
+	// The access rule type that can be "ip", "cert" or "user".
+	AccessType string `json:"access_type,omitempty"`
+	// The value that defines the access that can be a valid format of IP, cert or user.
+	AccessTo string `json:"access_to,omitempty"`
+	// The access credential of the entity granted share access.
+	AccessKey string `json:"access_key,omitempty"`
+	// The access level to the share is either "rw" or "ro".
+	AccessLevel string `json:"access_level,omitempty"`
+	// The state of the access rule
+	State string `json:"state,omitempty"`
+	// The access rule ID.
+	ID string `json:"id"`
+}
+
+// ExtractGrantAccess will get the GrantAccess object from the commonResult
+func (r commonResult) ExtractGrantAccess() (*GrantAccessRes, error) {
```

Comment:
Per discussion in #381, this should be changed from `commonResult` to `GrantAccessResult`.